synapse/handlers/user_directory.py

```diff
@@ -97,6 +108,23 @@ async def search_users(
         """
         results = await self.store.search_user_dir(user_id, search_term, limit)
 
+        merged_results = results["results"]
+        existing_user_ids = [user["user_id"] for user in merged_results]
```

Comment:
merging the external user directory with synapse's works so that: if user_id match -> user match.
I don't know if it is realistic that the display name or avatar is different in the "backend", but this seems like the safe(r) thing to do